ams/server/src/main/java/com/netease/arctic/server/ArcticServiceContainer.java

```diff
@@ -289,6 +298,32 @@ private void validateConfig(Map<String, Object> systemConfig) {
         throw new RuntimeException(
             String.format("Illegal terminal implement: %s, local, kyuubi, custom is available", terminalBackend));
       }
+
+      validateThreadCount(systemConfig, ArcticManagementConf.REFRESH_TABLES_THREAD_COUNT);
+      validateThreadCount(systemConfig, ArcticManagementConf.OPTIMIZING_COMMIT_THREAD_COUNT);
+
+      if (enabled(systemConfig, ArcticManagementConf.EXPIRE_SNAPSHOTS_ENABLED)) {
+        validateThreadCount(systemConfig, ArcticManagementConf.EXPIRE_SNAPSHOTS_THREAD_COUNT);
+      }
+
+      if (enabled(systemConfig, ArcticManagementConf.CLEAN_ORPHAN_FILES_ENABLED)) {
+        validateThreadCount(systemConfig, ArcticManagementConf.CLEAN_ORPHAN_FILES_THREAD_COUNT);
+      }
+      if (enabled(systemConfig, ArcticManagementConf.SYNC_HIVE_TABLES_ENABLED)) {
+        validateThreadCount(systemConfig, ArcticManagementConf.SYNC_HIVE_TABLES_THREAD_COUNT);
+      }
+    }
+    
+    private boolean enabled(Map<String, Object> systemConfig, ConfigOption<Boolean> config) {
+      return (boolean) systemConfig.getOrDefault(config.key(), config.defaultValue());
+    }
+
+    private void validateThreadCount(Map<String, Object> systemConfig, ConfigOption<Integer> config) {
+      int threadCount = (int) systemConfig.getOrDefault(config.key(), config.defaultValue());
+      if (threadCount <= 0) {
+        throw new RuntimeException(
```

Comment:
Is `IllegalArgumentException` better here?

Reply:
Yes, I fixed all of them when AMS configurations are illegal.